src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -720,69 +734,53 @@ def calculate_initial_medication_coverage_probabilities(
 
-        pop[data_values.COLUMNS.SBP_MEDICATION_START_DATE] = pd.NaT
-        pop[data_values.COLUMNS.LDLC_MEDICATION_START_DATE] = pd.NaT
-        pop[data_values.COLUMNS.DISCONTINUED_SBP_MEDICATION] = False
-        pop[data_values.COLUMNS.DISCONTINUED_LDLC_MEDICATION] = False
+        pop[start_date_col] = pd.NaT
+        pop[discontinued_col] = False
 
         sim_start = self.clock() + self.step_size()
 
         # Uniformly distribute medication start dates between 0-3 years in the past
-        ## sbp medication
-        sbp_medicated_idx = pop[
-            pop[data_values.COLUMNS.SBP_MEDICATION]
-            != data_values.SBP_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION
-        ].index
-        pop.loc[
-            sbp_medicated_idx,
-            data_values.COLUMNS.SBP_MEDICATION_START_DATE,
-        ] = sim_start - self.randomness.get_draw(
-            index=sbp_medicated_idx,
-            additional_key="initialized_sbp_medication_start_date",
-        ) * pd.Timedelta(
-            days=365.25 * data_values.INITIALIZATION_MEDICATON_START_DATE_YEARS_IN_PAST
-        )
-
-        ## ldlc medication
-        ldlc_medicated_idx = pop[
-            pop[data_values.COLUMNS.LDLC_MEDICATION]
-            != data_values.LDLC_MEDICATION_LEVEL.NO_TREATMENT.DESCRIPTION
-        ].index
+        medicated_idx = pop[pop[medication_col] != no_treatment_description].index
         pop.loc[
-            ldlc_medicated_idx,
-            data_values.COLUMNS.LDLC_MEDICATION_START_DATE,
+            medicated_idx,
+            start_date_col,
         ] = sim_start - self.randomness.get_draw(
-            index=ldlc_medicated_idx,
-            additional_key="initialized_ldlc_medication_start_date",
+            index=medicated_idx,
+            additional_key=f"initialize_{start_date_col}",
         ) * pd.Timedelta(
             days=365.25 * data_values.INITIALIZATION_MEDICATON_START_DATE_YEARS_IN_PAST
         )
 
         # Initialize medication discontinuation
-        ## sbp medication
-        not_sbp_medicated_idx = pop.index.difference(sbp_medicated_idx)
-        discontinued_sbp_idx = self.randomness.filter_for_probability(
-            population=not_sbp_medicated_idx,
-            probability=data_values.MEDICATION_DISCONTINUATION_PROBABILITY,
-            additional_key="initialize_discontinued_sbp_medication",
-        )
-        pop.loc[discontinued_sbp_idx, data_values.COLUMNS.DISCONTINUED_SBP_MEDICATION] = True
-
-        ## ldlc medication
-        not_ldlc_medicated_idx = pop.index.difference(ldlc_medicated_idx)
-        discontinued_ldlc_idx = self.randomness.filter_for_probability(
-            population=not_ldlc_medicated_idx,
-            probability=data_values.MEDICATION_DISCONTINUATION_PROBABILITY,
-            additional_key="initialize_discontinued_ldlc_medication",
+        not_medicated_idx = pop.index.difference(medicated_idx)
+        scaling_factor = self.medication_coverage_scaling_factors(not_medicated_idx)[
```

Comment:
All of the logic in this is the same as before except for anything to do w/ this `scaling_factor`. The fundamental change is that `probs` down below is now 0.314 / `probs` (instead of just 0.314 constant)